### ISSUE TITLE
Message "Error building: Stream was too long." when building "Tokimeki Memorial II"

### ISSUE THREAD

**Issue by LYK1003** [LiEnby/chovy-sign#21]
![ChovySign_v2_u6EjR5g6EY](https://github.com/LiEnby/chovy-sign/assets/67545948/db47c847-ffb1-4a43-bb59-1925fd1cefa6)

Before Tokimeki Memorial II, I have tried building for some PS1 games like Spyro 2 (Japanese Version) and Yu-Gi-Ou True Duel Monsters: Sealed Memories (Japanese Version) and they works great, even with PocketStation. The reasons of using Chovy Sign V2 method for some Japanese PS1 games instead of using the the Adrenaline one which I did most of the time is simply because I want to take advantage of using the PocketStation feature which Adrenaline Bubble's Manager lacks. 

Originally, there is Tokimeki Memorial II from PKGJ store and I tried to download it with the help of NoPspEmuDrm plugin so that I can install it right to the Vita home screen. However, when I finished downloading the game and tried to boot it, the error code "E-ffffffff" pop out and boot me back to the LiveArea. Since I already in 3.65 with enso installed so that there is no way to use the mail dump method to do the job.

Therefore, I tried to use Chovy Sign V2 to see if I can deal with the issue. "Tokimeki Memorial 2" is a huge game for PS1 game standard (2.14 GB) which contains 5 discs in total. After clicking "Go" button once I set up, the compression and encryption for 5 discs goes without any issue. But as soon as it gets into the psvimg section, the error "Error building: Stream was too long." kicks in which freezes the whole building process with no recovery at all. And I feel like this is because the total amount of storage for 5 discs is too huge which may results in this.

I wonder if there is a fix for this issue overall as it seems none of the users in general by far experiences this.

Thank you.

**Comment by LiEnby** [comment]
apparently there is a limit to how big a MemoryStream can be ..
https://stackoverflow.com/questions/53193085/memory-limitted-to-about-2-5-gb-for-single-net-process
and like, i did not know that-

**Comment by LYK1003** [comment]
Yeah, that would be a help if there is a way to deal with it. For now, I just buy the official game from PSN Japan and launch it with no issue. And this is by now THE ONLY WAY to play "Tokimeki Memorial 2" while taking advantage of PocketStation feature. 

Once again, thanks for your program so that I can inject many potential PS1 games with PocketStation feature :D

**Comment by LiEnby** [comment]
> Yeah, that would be a help if there is a way to deal with it. For now, I just buy the official game from PSN Japan and launch it with no issue. And this is by now THE ONLY WAY to play "Tokimeki Memorial 2" while taking advantage of PocketStation feature.
> 
> Once again, thanks for your program so that I can inject many potential PS1 games with PocketStation feature :D

huh? the official game works??
can i get the PKG file to that copy of the game ?

**Comment by LYK1003** [comment]
Sorry for late reply

https://drive.google.com/file/d/1Jf7kNe-aQRt2i8nuZs7K-4PH-xuiVoRm/view?usp=sharing

**(NOTE: YOU NEED TO TICK THE "HIDDEN ITEMS" OPTION TO SHOW CERTAIN HIDDEN FILES ON WINDOWS 10)**

Here is a little info regarding the rom:

The game id for this game is SLPM86355, which is "Tokimeki Memorial 2 [KONAMI THE BEST]"
Since this is a re-release for PSP so this is under the PSN Eboot structure and the Game ID for this on PSP is NPJJ00307

Let me know if there is any missing file, Thanks :D

Rom folder (SLPM86355):
"PSP/GAME"

Rif files (Game ID: NPJJ00307):
"PSP/LICENSE"

Save folder (SLPM86355):
"PSP/SAVEDATA"

**Comment by LiEnby** [comment]
> Failed to find versionkey for JP0101-NPJJ00307_00-0000000000000001

well theirs the problem ..

**Comment by LiEnby** [comment]
alright well i fixed it in NoPspEmuDrm; https://github.com/LiEnby/NoPspEmuDrm/releases/tag/v1.04

**Comment by LiEnby** [comment]
> apparently there is a limit to how big a MemoryStream can be .. https://stackoverflow.com/questions/53193085/memory-limitted-to-about-2-5-gb-for-single-net-process and like, i did not know that-
> 
> ive been meaning to make a option to use temporary files to be less memory-heavy anyway, probably when i actually work on this again

finally got around to it, this should work if you change the setting from MemoryStream to FileStream ..
